### ISSUE TITLE
Replace Zombienet with Chopsticks

### ISSUE THREAD

**Issue by NachoPal** [AcalaNetwork/chopsticks#657]
I am trying to replace Zombient with Chopsticks as a local testnet, but unfortunately I am facing some issues/limitations that make it not possible yet.

There are two different ways of using Chopsticks to deploy a network:
### 1. Running chains from genesis files
- **Problem:** #634 
- **Solution:** #635 
### 2. Forking production state.
- **Problem:** Runtime upgrades are potentially very slow, and predicting their completion duration is not possible.
- **Solutions:**
  -  Having a local node with minimal latency for the RPC storage requests
    - It mitigates the problem, but runtimes upgrades could still take too long

**Comment by xlc** [comment]
you could try use https://github.com/open-web3-stack/parachain-launch to generate genesis with hrmp channel opened

the block production shouldn’t be too slow with runtime upgrade. one reason could make it slow is migration modifying a lot of data

**Comment by ermalkaleci** [comment]
https://github.com/AcalaNetwork/chopsticks/pull/641
this will make migration run faster if you have specified a cache db

**Comment by NachoPal** [comment]
Sorry, I posted the issue by mistake before completion. Please check it again.

**Comment by NachoPal** [comment]
> one reason could make it slow is migration modifying a lot of data

Yes, exactly, that is the problem. I am trying to automate the testing process, but I can not really continue the testing flow after a runtime upgrade.

> Add some kind of event I can subscribe to know when the migration has ended to resume running my tests
This might do the trick for a simple solution

**Comment by NachoPal** [comment]
> - Block should not be produced until migration is done.
      - While it doesn't resolve the problem of slow migrations, at least I can automate the process of running the tests.

I think this is probably the easiest and optimal solution.

**Comment by ermalkaleci** [comment]
not sure what you mean. block is not produced before migration
